js/views/file_input_view.js

```diff
@@ -103,11 +106,15 @@
                     var blob;
                     do {
                         i = i - 1;
-                        // dataURLtoBlob -> components/blueimp-canvas-to-blob
-                        blob = dataURLtoBlob(
+                        // TODO: Replace with native `Canvas::toBlob`:
+                        // https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob
+                        blob = window.dataURLToBlobSync(
                             canvas.toDataURL('image/jpeg', quality)
                         );
                         quality = quality * maxSize / blob.size;
+                        // NOTE: During testing with a large image, we observed the
+                        // `quality` value being > 1. Should we clamp it to [0.5, 1.0]?
```

Comment:
The documentation says that the input should be between 0 and 1. Doesn't say what happens if we let it go above 1. I think we should clamp it.

Reply:
As discussed, we should re-test this whole function as part of #2026 and address it then.

Reply:
Done 👍